vm/systemSmartContracts/delegation.go

```diff
@@ -808,3 +1245,120 @@ func (d *delegation) IsContractEnabled() bool {
 func (d *delegation) IsInterfaceNil() bool {
 	return d == nil
 }
+
+func getTransferBackFromVMOutput(vmOutput *vmcommon.VMOutput) *big.Int {
+	transferBack := big.NewInt(0)
+	for _, outAcc := range vmOutput.OutputAccounts {
+		for _, outTransfer := range outAcc.OutputTransfers {
+			transferBack.Add(transferBack, outTransfer.Value)
+		}
+	}
+
+	return transferBack
+}
+
+func moveNodeFromList(sndList []*NodesData, dstList []*NodesData, key []byte) {
+	var toMoveNodeData *NodesData
+	for i, nodeData := range sndList {
+		if bytes.Equal(nodeData.BLSKey, key) {
+			copy(sndList[i:], sndList[i+1:])
+			toMoveNodeData = nodeData
+			break
+		}
+	}
+
+	dstList = append(dstList, toMoveNodeData)
+}
+
+func isSuccessReturnData(returnData []byte) bool {
+	if bytes.Equal(returnData, []byte{waiting}) {
+		return true
+	}
+	if bytes.Equal(returnData, []byte{ok}) {
+		return true
+	}
+	return false
+}
+
+func getSuccessAndUnSuccessKeys(returnData [][]byte, blsKeys [][]byte) ([][]byte, [][]byte) {
+	if len(returnData) == 0 {
+		return blsKeys, nil
```

Comment:
so we are assuming that if the return data is empty, everything succeeded

Reply:
yes. this is the implementation of the auction sc.